frontend/src/metabase/entities/schemas.js

```diff
@@ -65,12 +65,28 @@ export default createEntity({
 
     if (type === Questions.actionTypes.UPDATE) {
       const { question } = payload;
-      const schema = getCollectionVirtualSchemaId(question.collection);
-      if (!state[schema]) {
+      const schemaId = getCollectionVirtualSchemaId(question.collection);
+
+      const virtualQuestionId = getQuestionVirtualTableId(question);
+      const previousSchemaContainingTheQuestion = getPreviousSchemaContainingTheQuestion(
+        state,
+        schemaId,
+        virtualQuestionId,
+      );
+
+      if (previousSchemaContainingTheQuestion) {
+        state = removeVirtualQuestionFromSchema(
```

Comment:
We need to remove a question/model from an old collection after the question/model is moved away.